src/scheduler/workers/unpublish/dayBefore/sendDayBeforeEmails.ts

```diff
@@ -0,0 +1,54 @@
+import { findNonRespondentsForList } from "./findNonRespondentsForList";
+import { sendDayBeforeProviderConfirmation } from "./sendDayBeforeProviderConfirmation";
+import { sendDayBeforePostConfirmation } from "./sendDayBeforePostConfirmation";
+import { getMetaForList } from "../getMetaForList";
+import { schedulerLogger } from "scheduler/logger";
+import { ListWithCountryName } from "../../types";
+import { ListJsonData } from "server/models/types";
+
+export async function sendDayBeforeEmails(list: ListWithCountryName) {
+  const logger = schedulerLogger.child({ listId: list.id, method: "sendDayBeforeEmails" });
+
+  const meta = getMetaForList(list);
+  if (!meta) {
+    return;
+  }
+
+  if (meta.daysUntilUnpublish !== 1) {
+    logger.info(
+      `${meta.daysUntilUnpublish} does not match 1 day before unpublish, skipping sendDayBeforeProviderConfirmation and sendUnpublishPostConfirmation`
+    );
+    return;
+  }
+
+  const listItems = await findNonRespondentsForList(list);
+
+  if (!listItems.length) {
+    return;
+  }
+
+  logger.info(`sending provider email for list items ${listItems.map((listItem) => listItem.id)}`);
+  const providerEmailTasks = listItems.map(async (listItem) => await sendDayBeforeProviderConfirmation(listItem, meta));
+  const emailsForProviders = await Promise.allSettled(providerEmailTasks);
+  logger.info(
+    `Sent ${emailsForProviders.filter((promise) => promise.status === "fulfilled").length} provider emails for list ${
+      list.id
+    }`
+  );
+
+  let emailsSent = emailsForProviders;
+  const listJsonData = list.jsonData as ListJsonData;
+
+  // email post
+  if (listJsonData.users) {
+    const postEmailTasks = listJsonData.users.map(
+      async (emailAddress) => await sendDayBeforePostConfirmation(emailAddress, list, listItems.length, meta)
+    );
+    const emailsForPost = await Promise.allSettled(postEmailTasks);
+    emailsSent = emailsSent.concat(emailsForPost);
+    logger.info(
+      `Sent ${emailsForPost.filter((promise) => promise.status === "fulfilled").length} post emails for list ${list.id}`
+    );
+  }
+  return await Promise.allSettled(emailsSent);
```

Comment:
`return await Promise.allSettled([...emailsForProviders, ...emailsForPost])`

(instead of using concat)

FWIW - it looks like by the time this line is hit, you've already awaited on `allSettled` for both promise arrays. Was that your intent? 

I think both types of emails can be fired off at the same time (i.e. you don't need to wait for settled until the very end)

Reply:
I've addressed this with 

```
  // email post
  if (listJsonData.users) {
    const emailsForPost = listJsonData.users.map(
      async (emailAddress) => await sendUnpublishPostConfirmation(emailAddress, list, listItems.length, meta)
    );
    return await Promise.allSettled([...emailsForProviders, ...emailsForPost]);
  }
  return await Promise.allSettled(emailsForProviders);
```

as ts doesn't like this preferred version (Variable 'emailsForPost' implicitly has an 'any[]' type)
```
  let emailsForPost = [];
  if (listJsonData.users) {
    emailsForPost = listJsonData.users.map(
      async (emailAddress) => await sendUnpublishPostConfirmation(emailAddress, list, listItems.length, meta)
    );
  }
  return await Promise.allSettled([...emailsForProviders, ...emailsForPost]);
```

any thoughts?

Reply:
comment above relates to day not dayBefore